pkg/analyzer/nodes/node.go

```diff
@@ -0,0 +1,211 @@
+package nodes
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	clair "github.com/stackrox/scanner"
+	"github.com/stackrox/scanner/database"
+	"github.com/stackrox/scanner/pkg/analyzer"
+	"github.com/stackrox/scanner/pkg/component"
+	"github.com/stackrox/scanner/pkg/matcher"
+	"github.com/stackrox/scanner/pkg/metrics"
+	"github.com/stackrox/scanner/singletons/requiredfilenames"
+)
+
+// Information about extracted files.
+type fileMetadata struct {
+	// If true, the file has executable permissions.
```

Comment:
Is this also a regular file?

Reply:
[Yes](https://github.com/stackrox/scanner/blob/be36909287799a966506007287938218162d874c/pkg/matcher/matcher.go#L204).